web/db/actions/Post.ts

```diff
@@ -83,23 +90,46 @@ async function getResizedUploadUrl(uuid: string): Promise<string> {
 }
 
 async function deleteAttachments(keysToDelete: string[]) {
-  //will make key splitting more efficient later
-  const objectsToDelete = keysToDelete.map((keyToDelete) => ({
-    Key:
-      keyToDelete.split("/")[keyToDelete.split("/").length - 2] +
-      "/" +
-      keyToDelete.split("/")[keyToDelete.split("/").length - 1],
-  }));
-  const deleteObjectsCommand = new DeleteObjectsCommand({
-    Bucket: consts.storageBucket,
-    Delete: { Objects: objectsToDelete },
-  });
-  try {
-    const returned = await storageClient.send(deleteObjectsCommand);
-    return { success: true };
-  } catch (error) {
-    return { success: false };
+  for (let i = 0; i < 2; i++) {
+    const objectsToDelete = keysToDelete.map((keyToDelete) => ({
+      Key: keyToDelete,
+    }));
+    const deleteObjectsCommand = new DeleteObjectsCommand({
+      Bucket: consts.storageBucket,
+      Delete: { Objects: objectsToDelete, Quiet: true },
+    });
+    try {
+      const returned = await storageClient.send(deleteObjectsCommand);
+      let arr: string[] = [];
+      if (returned["Deleted"]) {
+        returned["Deleted"].forEach((element) => {
+          element["Key"] &&
+            !element["DeleteMarker"] &&
+            arr.push(element["Key"]);
+        });
+        keysToDelete = arr;
```

Comment:
This logic is a bit hard to follow - you shouldn't need to reset `keysToDelete` based on which items were deleted.

Per https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/classes/deleteobjectscommand.html:
> For each key, Amazon S3 performs a delete action and returns the result of that delete, success, or failure, in the response. **Note that if the object specified in the request is not found, Amazon S3 returns the result as deleted.**

Consider simplifying this to just check the number of deleted elements to determine whether we need to retry or not.